src/Features/LanguageServer/Protocol/Handler/References/FindAllReferencesHandler.cs

```diff
@@ -50,14 +50,15 @@ public FindAllReferencesHandler(IMetadataAsSourceFileService metadataAsSourceFil
             var position = await document.GetPositionFromLinePositionAsync(
                 ProtocolConversions.PositionToLinePosition(referenceParams.Position), cancellationToken).ConfigureAwait(false);
 
-            var context = new FindUsagesLSPContext(document, position, _metadataAsSourceFileService, cancellationToken);
+            var context = new FindUsagesLSPContext(
+                referenceParams.PartialResultToken, document, position, _metadataAsSourceFileService, cancellationToken);
 
             // Finds the references for the symbol at the specific position in the document, reporting them via streaming to the LSP client.
-            // TODO: Change back FAR to use streaming once the following LSP bug is fixed:
-            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1094786/
             await findUsagesService.FindReferencesAsync(document, position, context).ConfigureAwait(false);
+            await context.OnCompletedAsync().ConfigureAwait(false);
 
-            return context.GetReferences().ToArray();
+            // The results have already been reported to the client, so we don't need to return anything here.
+            return Array.Empty<LSP.VSReferenceItem>();
```

Comment:
i may have missed it.  but if you're chunking, what makes sure any final items still get sent?

Reply:
The [OnCompletedAsync()](https://github.com/dotnet/roslyn/pull/45013/files#diff-cc27c15ba5f5bd3a882e739859f933e2R77) method sends out any remaining references that weren't reported via chunking.